src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,239 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../../TodosContext';
+import * as todoService from '../../api/todos';
+import { Filter } from '../../types/Filter';
+import { UserWarning } from '../../UserWarning';
+import { TodosList } from '../TodosList';
+import { TodosFilter } from '../TododsFilter';
+import { TempTodoContext } from '../../TempTodoContext';
+import { ErrorContext } from '../../ErrorContext';
+import { Error } from '../../types/ErrorMessage';
+
+const USER_ID = 11903;
+const ERROR_DELAY = 3000;
+
+export const TodoApp: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const { tempTodo, setTempTodo } = useContext(TempTodoContext);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [status, setStatus] = useState('all');
+  const { errorMessage, setErrorMessage } = useContext(ErrorContext);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const refInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(Error.LOAD_TODOS);
+      })
+      .finally(() => {
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+      });
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (status) {
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+
+  const completedTodo = todos.some(todo => todo.completed);
+
+  const addTodo = () => {
+    todoService.createTodo({
+      userId: USER_ID,
+      title: todoTitle.trim(),
+      completed: false,
+    })
+      .then(newTodo => {
+        setTodos([...todos, newTodo]);
+        setTodoTitle('');
+        setTempTodo(null);
+      })
+      .catch((error) => {
+        setErrorMessage(Error.ADD_TODO);
+        setTempTodo(null);
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+        throw error;
```

Comment:
do not throw it - it'll lead to page crash. You caught error and handled no need to throw it again